book/src/developers/designs/identity-verification.md

```diff
@@ -0,0 +1,160 @@
+ ││  Server 3  │◀────────────▶│  Server 4  ││
+ ││            │              │            ││
+ ││            │              │            ││
+ ││            │              │            ││
+ │└────────────┘              └────────────┘│
+ │                                          │
+ └──────────────────────────────────────────┘
+```
+
+## User flows:
+
+### Disclaimer: the following user flows assume that the servers have already established a trust bond.
+
+- ### Servers belonging to the same domain:
+
+  Since the servers belong to the same domain, it will be safe to replicate the database between them, and therefore all the pub/secrets keys of every person will be available on both servers.
+  Let's suppose that Alice and Bob want to mutually authenticate each other, then the following steps will be performed:
+
+  1. Alice shares her SPN with Bob and asks for Bob's SPN
+  1. Bob receives Alice's SPN and replies by sending his SPN
+  1. Both Alice and Bob insert the acquired information in their Kanidm client.
+  1. Their servers compute a HMAC hash using the secret key and the _token-time_ associated with the server identified by the domain name provided by the user, and then to shorten it to ~6 digits
+  1. The servers will then encrypt the hash using the private key of the respective users.
+  1. The encrypted hash will be transformed in a more human readable format for the users, so that the users will be able to easily communicate it to each other, ie in a phone call or in a chat.
+  1. The users will then decrypt the message using the public key associated with the SPN they received, and will compare the decrypted hash with the one computed by their server.
+
+  ```
+                  ┌──────────────────────────────────────────┐
+                  │Domain A                                  │
+                  │┌────────────┐              ┌────────────┐│
+                  ││            │              │            ││
+                  ││            │              │            ││
+  ┌──────────┐    ││            │              │            ││    ┌──────────┐
+  │  Alice   │    ││  Server 1  │◀────────────▶│  Server 2  ││    │   Bob    │
+  └──────────┘    ││            │              │            ││    └──────────┘
+                  ││            │              │            ││
+                  ││            │              │            ││
+                  │└────────────┘              └────────────┘│
+                  │                                          │
+                  └──────────────────────────────────────────┘
+  ```
+
+- ### Servers belonging to different domains:
+  This case is very similar to the previous one, except that the servers don't belong to the same domain, and therefore the database is not replicated between them. In this case, the servers will stil have to share all the users' public keys, but the way that will be achieved is (for now) out of the scope of this document.
```

Comment:
Well, trusts over a domain is actually just a special case of replication where we replicate a partial entry to the partner domain. We can just send the public part of the ec key in the partial entry over the trust boundary so here server 1 has Alice's private key and public key, and would have bob's public key only. Inverse, Server 4 would have Alices public key, and bobs private and public key. 

So again, they can each use what they have (alice's private + bobs public || bobs private + alices public) for the ecdh operation and arrive at the same secret, again without direction communication or contact.